bigquery_etl/glam/scalar_aggregates_incremental.py

```diff
@@ -1,68 +1,221 @@
+    env = Environment(loader=PackageLoader("bigquery_etl", "glam/templates"))
+    init_sql = env.get_template("clients_scalar_aggregates_v1.init.sql")
+    return reformat(
+        init_sql.render(
+            header=header,
+            destination_table=destination_table,
+            attributes_type=",".join(
+                f"{name} {dtype}" for name, dtype in zip(attributes, attributes_type)
+            ),
+            user_data_type=user_data_type,
+            partition_clause=partition_clause,
+        )
+    )
+
+
+def telemetry_variables():
+    """Telemetry specific variables."""
+    return dict(
+        user_data_type="""
+            ARRAY<
+                STRUCT<
+                metric STRING,
+                metric_type STRING,
+                key STRING,
+                process STRING,
+                agg_type STRING,
+                value FLOAT64
+                >
+            >
+        """,
+        user_data_attributes=[
+            "metric",
+            "metric_type",
+            "key",
+            "process",
+            # [agg_type, value] are shared between telemetry and glean
+        ],
+        attributes=["client_id", "os", "app_version", "app_build_id", "channel"],
+        attributes_type=["STRING", "STRING", "INT64", "STRING", "STRING"],
+        extract_select_clause=f"""
+            * EXCEPT(app_version),
+            CAST(app_version AS INT64) as app_version
+        """,
+        join_filter="""
+            LEFT JOIN
+                latest_versions
+            USING
+                (channel)
+            WHERE
+                app_version >= (latest_version - 2)
+        """,
+        partition_clause="""
+            PARTITION BY
+                RANGE_BUCKET(app_version, GENERATE_ARRAY(30, 200, 1))
+        """,
+    )
+
+
+def glean_variables():
+    """Glean/Fenix specific variables."""
+    return dict(
+        # does not include "process" field
+        user_data_type="""
+            ARRAY<
+                STRUCT<
+                metric STRING,
+                metric_type STRING,
+                key STRING,
+                agg_type STRING,
+                value FLOAT64
+                >
+            >
+        """,
+        user_data_attributes=[
+            "metric",
+            "metric_type",
+            "key",
+            # [agg_type, value] are shared between telemetry and glean
+        ],
+        attributes=[
+            "client_id",
+            "ping_type",
+            "os",
+            "app_version",
+            "app_build_id",
+            "channel",
+        ],
+        attributes_type=["STRING", "STRING", "STRING", "INT64", "STRING", "STRING"],
+        extract_select_clause="*",
+        # no filtering on channel
+        join_filter="",
+        # TODO: is this the appropriate partitioning of the table?
+        partition_clause="""
+            PARTITION BY
+                RANGE_BUCKET(app_version, GENERATE_ARRAY(0, 100, 1))
```

Comment:
I'm not sure what appropriate partitioning for the table is. Does it need partitioning and clustering at all?

Reply:
> I'm not sure what appropriate partitioning for the table is. Does it need partitioning and clustering at all?

The partitioning by version is useful because the join filter in desktop looks at `app_version >= (latest_version - 2)`.